### PR TITLE
Kernel:  add config fragment support

### PR DESCRIPTION
**Pull request by devimc**
original pull request https://github.com/kata-containers/packaging/pull/314

### CODE REVIEW

File: kernel/build-kernel.sh [hunk 1]
```diff
@@ -136,19 +138,93 @@ get_major_kernel_version() {
...
+# - arg2 - path to kernel sources
+#
+get_kernel_frag_path() {
+	local arch_path="$1"
+	local common_path="${arch_path}/../common"
+	local kernel_path="$2"
+	local cmdpath="${kernel_path}/scripts/kconfig/merge_config.sh"
+	local config_path="${arch_path}/.config"
+
+	local arch_configs="$(ls ${arch_path}/*.conf)"
```
Comment (jodh-intel) [inline kernel/build-kernel.sh]: Do we need arch-specific kernel fragment config files for all the other (non-`x86_64`) architectures to ensure those kernel continue to build as they do today?

Comment (grahamwhaley) [reply kernel/build-kernel.sh]: The code is written so if it does not find arch specific fragments it reverts to using the non-fragment configs. Thus, we can add arches as and when they have their fragments ready, and not delay the merge of the framework and x86 fragments :-)

File: kernel/build-kernel.sh [hunk 2]
```diff
@@ -136,19 +138,93 @@ get_major_kernel_version() {
...
+	local redefined_string="not in final"
+	local redundant_string="not in final"
+
+	# Later, if we need to add kernel version specific subdirs in order to
+	# handle specific cases, then add the path definition and search/list/cat
+	# here.
+	local all_configs="${common_configs} ${arch_configs}"
+
+	info "Constructing config from fragments: ${config_path}"
+	local results=$(export KCONFIG_CONFIG=${config_path}; export ARCH=${arch_target}; cd ${kernel_path}; ${cmdpath} -r -n ${all_configs})
```
[the remark this answers is not in the document]
Comment (devimc) [reply kernel/build-kernel.sh]: fixed

**Comment by devimc** [comment]
@egernst @grahamwhaley @ganeshmaharaj CI is happy, can we merge this?

**Comment by grahamwhaley** [comment]
@devimc 
- any progress/update on SRIOV support?
- @egernst - would we be happy merging without SRIOV - that would then become the default kernel config for Kata x86 releases...
- obs packaging CI looks to have failed

**Comment by jcvenegas** [comment]
All green - merged :) , kudos @devimc

**Pull request merged**